src/matter/interaction/InteractionServer.ts

```diff
@@ -224,30 +226,49 @@ export class InteractionServer implements ProtocolHandler<MatterDevice> {
         logger.debug(`Received subscribe request from ${exchange.channel.getName()}`);
 
         if (!exchange.session.isSecure()) throw new Error("Subscriptions are only implemented on secure sessions");
         const session = exchange.session as SecureSession<MatterDevice>;
         const fabric = session.getFabric();
         if (fabric === undefined) throw new Error("Subscriptions are only implemented after a fabric has been assigned");
 
+        if ((!attributeRequests || attributeRequests.length === 0) && (!eventRequests || eventRequests.length === 0)) {
+            await messenger.sendStatus(StatusCode.InvalidAction);
+            return;
+        }
+
         if (!keepSubscriptions) {
             session.clearSubscriptions();
         }
 
+        // TODO add eventsRequest and other missing supports
         if (attributeRequests !== undefined) {
             logger.debug(`Subscribe to ${attributeRequests.map(path => this.resolveAttributeName(path)).join(", ")}`);
-            let attributes = this.getAttributes(attributeRequests);
 
-            if (attributeRequests.length === 0) throw new Error("Invalid subscription request");
+            if (attributeRequests.length === 0) throw new Error("Unsupported subscription request with empty attribute list");
             if (minIntervalFloorSeconds < 0) throw new Error("minIntervalFloorSeconds should be greater or equal to 0");
             if (maxIntervalCeilingSeconds < 0) throw new Error("maxIntervalCeilingSeconds should be greater or equal to 1");
             if (maxIntervalCeilingSeconds < minIntervalFloorSeconds) throw new Error("maxIntervalCeilingSeconds should be greater or equal to minIntervalFloorSeconds");
 
-            const subscriptionId = session.addSubscription(SubscriptionHandler.Builder(session.getContext(), fabric, session.getPeerNodeId(), attributes, minIntervalFloorSeconds, maxIntervalCeilingSeconds));
+            let attributes = this.getAttributes(attributeRequests);
+
+            // TODO: Interpret specs:
+            // The publisher SHALL compute an appropriate value for the MaxInterval field in the action. This SHALL respect the following constraint: MinIntervalFloor ≤ MaxInterval ≤ MAX(SUBSCRIPTION_MAX_INTERVAL_PUBLISHER_LIMIT=60mn, MaxIntervalCeiling)
+
+            if (this.nextSubscriptionId === 0xFFFFFFFF) this.nextSubscriptionId = 0;
+            const subscriptionHandler = new SubscriptionHandler(this.nextSubscriptionId++, session.getContext(), fabric, session.getPeerNodeId(), attributes, minIntervalFloorSeconds * 1000, maxIntervalCeilingSeconds * 1000);
+            session.addSubscription(subscriptionHandler);
+            const subscriptionId = subscriptionHandler.subscriptionId;
```

Comment:
Use const subscriptionId = this.nextSubscriptionId++, to make this m,ore readable